authorization-common/src/main/java/com/sequenceiq/authorization/service/CommonPermissionCheckingUtils.java

```diff
@@ -18,39 +22,95 @@
 import org.springframework.security.access.AccessDeniedException;
 import org.springframework.stereotype.Component;
 
+import com.google.common.base.Joiner;
 import com.google.common.collect.Lists;
-import com.sequenceiq.authorization.annotation.DisableCheckPermissions;
+import com.google.common.collect.Maps;
 import com.sequenceiq.authorization.annotation.AuthorizationResource;
+import com.sequenceiq.authorization.annotation.DisableCheckPermissions;
 import com.sequenceiq.authorization.resource.AuthorizationResourceAction;
 import com.sequenceiq.authorization.resource.AuthorizationResourceType;
+import com.sequenceiq.authorization.resource.RightUtils;
 
 @Component
 public class CommonPermissionCheckingUtils {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(CommonPermissionCheckingUtils.class);
 
-    @Inject
-    private UmsAccountAuthorizationService umsAccountAuthorizationService;
+    private final UmsAccountAuthorizationService umsAccountAuthorizationService;
+
+    private final UmsResourceAuthorizationService umsResourceAuthorizationService;
+
+    private final Map<AuthorizationResourceType, DefaultResourceChecker> defaultResourceCheckers;
 
     @Inject
-    private UmsResourceAuthorizationService umsResourceAuthorizationService;
+    public CommonPermissionCheckingUtils(UmsAccountAuthorizationService umsAccountAuthorizationService,
+            UmsResourceAuthorizationService umsResourceAuthorizationService,
+            List<DefaultResourceChecker> defaultResourceCheckers) {
+        this.umsAccountAuthorizationService = umsAccountAuthorizationService;
+        this.umsResourceAuthorizationService = umsResourceAuthorizationService;
+        this.defaultResourceCheckers = Maps.immutableEnumMap(defaultResourceCheckers.stream()
+                .collect(toMap(DefaultResourceChecker::getResourceType, Function.identity())));
+    }
```

Comment:
Is there a reason to switch to constructor injection? I have no problem with it, just other classes in the module don't use it

Reply:
This way the field can be final and we don't need a silly `@PostConstruct`. Of course we can start a religious war like "follow the conventions", you could have done it from java config  or there is nothing wrong with non final fields and `@PostConstruct`. I think it's better to make it final and use an immutable EnumMap in the constructor.

Reply:
it is ok for me, but one day one team member will start that war, i'm almost 100% sure about that :) I'm just like if a module follows one direction regarding things like this injection stuff based on team agreement, but I do not want to start (or getting involved in) a long discussion about this. Anyway I suggest to involve at least one other team member to check this, because some people do not like this, because I know there were discussions regarding this topic.